jax/_src/interpreters/partial_eval.py

```diff
@@ -1810,13 +1811,17 @@ def _const_folding_and_forwarding(
   new_eqns = []
   for eqn in jaxpr.eqns:
     # always apply invar substitutions
-    eqn = eqn.replace(invars=[var_subs.get(v, v) for v in eqn.invars])
+    eqn = eqn.replace(invars=[(var_subs.get(v, v)
```

Comment:
Maybe define a convenience function wrapping `var_subs.get(v, v)` with the proper type annotation, rather than putting the `isinstance` logic inline in the list comprenensions.

This is one of the arguments against type annotations that I have some sympathy for... it can sometimes lead to stilted code, especially when applied retroactively to existing implementations.

Reply:
> This is one of the arguments against type annotations that I have some sympathy for... it can sometimes lead to stilted code, especially when applied retroactively to existing implementations.

True, but I think as a reader who is unfamiliar with this code, I find it easier to understand with the annotations :smile: